ci_requirements/all-3.12.txt

```diff
@@ -391,7 +391,7 @@ sortedcontainers==2.4.0
     # via hypothesis
 soupsieve==2.5
     # via beautifulsoup4
-sphinx==7.3.7
+sphinx==7.4.3
```

Comment:
📚 Seems likely that the Sphinx update is most likely to have caused the new doc build failures.